### PR TITLE
tentacles_manager integration 

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Commons/pull/28

### CODE REVIEW

File: octobot_trading/modes/abstract_trading_mode.py [hunk 1]
```diff
@@ -134,10 +119,8 @@ def is_backtestable() -> bool:
         raise NotImplementedError("create_consumers not implemented")
 
     def load_config(self) -> None:
-        config_file = self.get_config_file_name()
         # try with this class name
-        if os.path.isfile(config_file):
-            self.trading_config = load_config(config_file)
+        self.trading_config = get_tentacle_config(self.__class__)
```
Comment (Herklos) [inline octobot_trading/modes/abstract_trading_mode.py]: :+1:

File: tests/exchanges/test_exchange_factory.py [hunk 2]
```diff
@@ -18,30 +18,31 @@
...
+TEST_TRADING_TENTACLES_CONFIG_FILE_PATH = os.path.join(TEST_TRADING_TENTACLES_CONFIG_PATH, CONFIG_TENTACLES_FILE)
 
 
 def create_test_tentacles_config():
-    if not os.path.exists(TEST_TRADING_TENTACLES_PATH):
-        os.makedirs(TEST_TRADING_TENTACLES_PATH)
-        copyfile(os.path.join(TEST_CONFIG_FOLDER, CONFIG_TRADING_FILE), TEST_TRADING_TENTACLES_CONFIG_PATH)
+    if not os.path.exists(TEST_TRADING_TENTACLES_CONFIG_PATH):
+        os.makedirs(TEST_TRADING_TENTACLES_CONFIG_PATH)
+        #TODO
```
Comment (Herklos) [inline tests/exchanges/test_exchange_factory.py]: What's should be done here ?

Comment (GuillaumeDSM) [reply tests/exchanges/test_exchange_factory.py]: nothing, mistake

File: octobot_trading/api/modes.py [hunk 3]
```diff
@@ -43,16 +39,13 @@ def get_trading_mode_current_state(trading_mode) -> tuple:
     return trading_mode.get_current_state()
 
 
-def get_activated_trading_mode(config) -> AbstractTradingMode.__class__:
-    return util_get_activated_trading_mode(config)
+def get_activated_trading_mode(config, tentacles_setup_config) -> AbstractTradingMode.__class__:
```
Comment (Herklos) [inline octobot_trading/api/modes.py]: https://github.com/Drakkar-Software/OctoBot-Backtesting/pull/38#discussion_r399713679

File: octobot_trading/exchanges/exchange_builder.pxd [hunk 4]
```diff
@@ -43,6 +43,6 @@ cdef class ExchangeBuilder:
     cpdef ExchangeBuilder is_margin(self, bint use_margin=*)
     cpdef ExchangeBuilder is_exchange_only(self)
     cpdef ExchangeBuilder is_ignoring_config(self)
-    cpdef ExchangeBuilder use_trading_mode(self, str trading_tentacles_path)
+    cpdef ExchangeBuilder use_tentacles_setup_config(self, str tentacles_setup_config)
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_builder.pxd]: It's not a `str` but a `dict` right ?

File: octobot_trading/exchanges/exchange_builder.pxd [hunk 5]
```diff
@@ -30,7 +30,7 @@ cdef class ExchangeBuilder:
 
     cdef str _matrix_id
 
-    cdef str _trading_tentacles_path
+    cdef str _tentacles_setup_config
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_builder.pxd]: See below

File: octobot_trading/exchanges/exchange_builder.pxd (hunk 4 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/exchange_builder.pxd]: object but right

**Pull request merged**